nexus/social_media/services.py

```diff
@@ -0,0 +1,77 @@
+# Third Party Stuff
+import tweepy
+from django.conf import settings
+from django.utils import timezone
+
+# nexus Stuff
+from nexus.base import exceptions as exc
+from nexus.social_media.models import Post
+
+
+def update_post_object(post):
+    """Update post instance after the post is published.
+
+    :param post: Post model instance.
+
+    """
+    post.posted_time = timezone.now()
+    post.is_posted = True
+    post.save()
+
+
+def get_twitter_api_object(TWITTER_OAUTH):
+    """Function to generate a twitter api object.
+
+    :param TWITTER_OAUTH: A dictionary having essential twitter oauth tokens viz.
+    consumer_key, consumer_secret, access_key, access_secret.
+
+    :returns: twitter API object.
+
+    :raises WrongArguments: Exception, when invalid twitter oauth token(s) are provided.
+
+    """
+    try:
+        auth = tweepy.OAuthHandler(TWITTER_OAUTH['consumer_key'], TWITTER_OAUTH['consumer_secret'])
+        auth.set_access_token(TWITTER_OAUTH['access_key'], TWITTER_OAUTH['access_secret'])
+        twitter_api = tweepy.API(auth)
+        return twitter_api
+    except tweepy.error.TweepError:
+        raise exc.WrongArguments("TweepError: Invalid Twitter OAuth Token(s).")
+
+
+def post_to_twitter(post_id):
+    """Function to post on twitter.
+
+    :param post_id: UUID of the post instance to be posted.
+
+    :raises BadRequest: Exception, when unable to post to twitter using tweepy.
+
+    """
+    post = Post.objects.get(pk=post_id)
+    twitter_api = get_twitter_api_object(settings.TWITTER_OAUTH)
+
+    try:
+        if post.image:
+            filename = post.image.file.name
+            if post.text:
```

Comment:
Is there any error if `post.text` as `None` is passed to `twitter_api.update_with_media`?

Reply:
No, there is no error if `post.text` is passed to `twitter_api.update_with_media` as `None`.
Great! We may remove this condition and directly call `twitter_api.update_with_media`,

Reply:
You can remove all the conditions here.

Reply:
No, I can not remove all the conditions. Because `filename = post.image.file.name` will be possible only when `post.image` is not `None`, otherwise it will fail the task with `ValueError: The 'image' attribute has no file associated with it.`

Therefore, I could only remove one if-else condition that I have removed in my [last commit](https://github.com/CuriousLearner/nexus/pull/20/commits/eeb09a3556091dd07361fb4268fd9d604c019186).